src/dhcp6relay/src/relay.cpp

```diff
@@ -681,24 +680,26 @@ void dhcp6relay_stop()
  */
 void loop_relay(std::vector<relay_config> *vlans) {
     std::vector<int> sockets;
-    
-    for(std::size_t i = 0; i<vlans->size(); i++) {
-        struct relay_config config = vlans->at(i);
+    std::vector<relay_config *> configs;
+
+    for(relay_config vlan: *vlans) {
+        relay_config *config = new relay_config(vlan);
```

Comment:
So this memory allocation can be avoided by resuing the object that's already in `vlans` (this list will exist for the duration of the application, since this is actually allocated in the `main` method). To do this, use `relay_config *config = &vlans->at(i)`. That way, you don't need to free the memory at the end of this function.

Reply:
Thanks for pointing it out! Updated